New RNA-seq tutorial: de novo transcriptome reconstruction

malloryfreeberg opened the pull request:
Mo and I have put together a tutorial for de novo reconstruction of a transcriptome from RNA-seq data with a reference genome for alignment. The data used in this tutorial are from the same study we used for the ChIP-seq tutorial, so the tutorials are complementary. We are still working on slides and a Dockerfile, but the major tutorial items are ready. Please let us know if we're missing anything from the tutorial or metadata files.

malloryfreeberg commented:
This satisfies issue #197 so feel free to close that issue if this gets merged.

bgruening commented on metadata/RNA-Seq.yml:
```diff
@@ -56,6 +56,42 @@ material:
+        type: "internal"
+        link: "/NGS-mapping"
+    -
+        title: "IGV"
```
Trackster is a requirement as well, right? I have the feeling that not many people have a fully functional trackster these days. So maybe we should mention this.

malloryfreeberg replied:
This is also a remnant of copy+paste. Will update to Trackster since that is what we actually use in the tutorial.

bgruening commented on metadata/RNA-Seq.yml:
```diff
@@ -56,6 +56,42 @@ material:
+    - "Reconstruction of transcripts without reference transcriptome (de novo)"
+    - "Analysis of differentially expressed genes"
+
+    time_estimation: "1d"
```
Really a day? :) 6h maybe? Or does a day, really means 24h?

malloryfreeberg replied:
Remnant of copy+paste. I'll update to a more reasonable time :)

bgruening commented on metadata/RNA-Seq.yml:
```diff
@@ -56,6 +56,42 @@ material:
+    -
+        title: "IGV"
+        type: "internal"
+        link: "/Introduction/IGV"
```
Where are you using IGV?

malloryfreeberg replied:
We aren't. Changing to Trackster.

bgruening commented on RNA-Seq/tutorials/de_novo.md:
```diff
@@ -0,0 +1,377 @@
+For more information about `DESeq2` and its outputs, you can have a look at [`DESeq2` documentation](https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf).
+
+# Visualization
+Now that we have a list of transcript expression levels and their differential expression levels, it is time to visually inspect our transcript structures and the reads they were predicted from. It is a good practice to visually inspect (and present) loci with transcripts of interest. Fortuantely, there is a built-in genome browser in Galaxy, **Trackster**, that make this task simple (and even fu ...
```
Add hint for IGV and a link to the IGV tutorial?

malloryfreeberg replied:
I would argue just to keep everything in relation to Trackster for the tutorials. Is there a reason some tutorials use IGV? Why not UCSC or another browser?

MoHeydarian replied:
I agree with @malloryfreeberg, we should try to keep users inside the Galaxy environment for viewing their data on a genome browser. Moving data to an external genome browser introduces the opportunity to lose track of files and is extremely cumbersome (for larger datasets) IMO. Using Trackster within Galaxy makes viewing data a seamless step (when it works).

You raise a good point @bgruening, Trackster has not always been stable. It has been working more reliably over the past year. It is really a great feature within Galaxy to promote "looking at your data" with a couple of clicks. Maybe we could convince some folks participating in the GCC2017 hackathon to address Trackster issues. I'll be suggesting ways to make Trackster more usable at the GCC2017 datathon (specifically by making reference datasets and repeat tracks available as default).

bgruening replied:
It's just a feeling, but don't know about much Galaxy instances that offer a working trackster experience. IGV can be installed by everyone, very easily. I think we can leave this in, but maybe we should just add a sentence, a link to IGV or others (if we have) to just give pointers.

I totally agree that trackster would be the way to go it is actively maintained.

malloryfreeberg commented:
@bgruening I have double-checked the images and we had replaced 2 of them accidentally for the ref_based tutorial (DESeq2_PCA.png and DESeq2_heatmap.png). I have returned the original images, renamed the images for the de_novo tutorial (DESeq2_PCAplot.png and DESeq2_heatmap_G1E_mega.png), and everything looks good to me. Note that I believe the PCA and heatmap for the ref_based tutorial is no longer embedded in the current ref_bsed.md file, but they are embedded correctly in our older version of ref_based.md.

bgruening commented:
@malloryfreeberg There are a few broken links to zenodo, can you please check that. Otherwise, I'm fine with merging! Thanks a lot!

bgruening commented:
Nice! Great work @malloryfreeberg and @MoHeydarian!
